### PR TITLE
[SPARK-18121][SQL] Unable to query global temp views when hive support is enabled

### PR DESCRIPTION
**Pull request by skambha**
## What changes were proposed in this pull request?

Issue: 
Querying on a global temp view throws Table or view not found exception. 

Fix:
Update the lookupRelation in HiveSessionCatalog to check for global temp views similar to the SessionCatalog.lookupRelation.

Before fix: 
Querying on a global temp view ( for. e.g.:  select \* from global_temp.v1)  throws Table or view not found exception

After fix: 
Query succeeds and returns the right result. 
## How was this patch tested?
- Two unit tests are added to check for global temp view for the code path when hive support is enabled.
- Regression unit tests were run successfully. ( build/sbt -Phive hive/test, build/sbt sql/test, build/sbt catalyst/test)

### CODE REVIEW

**Comment by gatorsmile** [comment]
ok to test

**Comment by gatorsmile** [comment]
@cloud-fan Could you take a look at this? The code changes look good to me. Thanks!

**Comment by viirya** [comment]
LGTM

**Comment by cloud-fan** [comment]
ok to test

**Comment by cloud-fan** [comment]
LGTM

**Comment by gatorsmile** [comment]
retest this please

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala [hunk 1]
```diff
@@ -65,6 +65,20 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
   import hiveContext._
   import spark.implicits._
 
+  test("query global temp view") {
+    val df = Seq(1).toDF("i1")
+    df.createGlobalTempView("tbl1")
+    checkAnswer(spark.sql("select * from global_temp.tbl1"), Row(1))
```
Comment (skambha) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala]: Thanks @cloud-fan.  I have changed the test to get the global_temp db from the conf.   spark.conf.get("spark.sql.globalTempDatabase”)

It looks like there are two ways to get it: 
- One is using the conf
- Accessing the spark.sharedState.globalTempViewManager.database

In the test, I am now getting it from the conf.   Please see if this is ok.  Thanks.

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by skambha** [comment]
Thanks @cloud-fan, @gatorsmile, @viirya

**Pull request closed without merging**